spec/components/base_component_spec.rb

```diff
@@ -1,11 +1,13 @@
 require 'rails_helper'
 
 RSpec.describe BaseComponent, type: :component do
+  # rubocop:disable RSpec/LeakyConstantDeclaration
   class ExampleComponent < BaseComponent
```

Comment:
I opted to allow these instead of turn them into `let`s because I got errors like this

```
       undefined method `demodulize' for nil:NilClass
```

Other options
- `stub_const('ExampleComponent', ...)`
- making sure to define a `self.name` method on these anonymous classes